h2/src/main/org/h2/command/dml/ScriptCommand.java

```diff
@@ -318,12 +318,7 @@ public int compare(Table t1, Table t2) {
             // Generate CREATE CONSTRAINT ...
             final ArrayList<SchemaObject> constraints = db.getAllSchemaObjects(
                     DbObject.CONSTRAINT);
-            Collections.sort(constraints, new Comparator<SchemaObject>() {
-                @Override
-                public int compare(SchemaObject c1, SchemaObject c2) {
-                    return ((Constraint) c1).compareTo((Constraint) c2);
-                }
-            });
+            Collections.sort(constraints, null);
```

Comment:
shouldn't this be 
    Collections.sort(constraints) 
?

Reply:
It does not matter at all, at least in modern versions of Java they do exactly the same thing. But I can change it to avoid such questions in the future.

Reply:
Oh, this is not possible without an ugly explicit cast. Single-argument method require `Collection<T extends Comparable<? super T>>` unlike this methods that accepts any collection. We should perform an unsafe cast to `ArrayList<Constraint>` and suppress the warning to use this method.

Reply:
then just leave it as is